middleware.js

```diff
@@ -53,7 +53,19 @@ module.exports = function(compiler, options) {
 	if(typeof options.reporter !== "function") options.reporter = defaultReporter;
 
 	// store our files in memory
-	var fs = compiler.outputFileSystem = new MemoryFileSystem();
+	var fs;
+	if(options.fileSystem) {
```

Comment:
Okay, could you remove `options.fileSystem` then? After that I'll do some tests and merge this in.